spec/features/demonstrating_nesting_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'spec_helper'
+require 'samvera/nesting_indexer'
+require 'samvera/nesting_indexer/exceptions'
+require 'samvera/nesting_indexer/adapters'
+require 'support/feature_spec_support_methods'
+
+# :nodoc:
+module Samvera
+  module NestingIndexer
+    RSpec.describe 'Demonstrating nesting' do
+      include Support::FeatureSpecSupportMethods
+      before do
+        NestingIndexer.adapter.clear_cache!
+      end
+
+      # Check if a document can be nested within itself.
+      def allowed_to_nest?(child_id:, parent_id:)
+        return false if child_id == parent_id # quick short circuit; don't let a parent be it's own child.
+        parent = NestingIndexer.adapter.find_index_document_by(id: parent_id)
+        # The more complicated test case. Check the parent such that the given child is not part of the parent's ancestry.
+        return false if parent.ancestors.detect { |pathname| pathname.include?(child_id) }
+        true
+      end
+
+      def verify_allowed_parent_ids(id:, allowed_parent_ids:)
+        actual_allowed = []
+        NestingIndexer.adapter.each_index_document do |document|
+          actual_allowed += [document.id] if allowed_to_nest?(child_id: id, parent_id: document.id)
+        end
+        expect(actual_allowed.sort).to eq(allowed_parent_ids.sort)
+      end
+
+      context 'querying what other documents in which the given document can nest' do
+        [
+          {
+            label: "a simple non-nested graph",
+            starting_graph: { parent_ids: { a: [], b: [], c: [] } },
+            expectations: [
+              { id: 'a', allowed_parent_ids: ['b', 'c'] },
+              { id: 'b', allowed_parent_ids: ['a', 'c'] },
+              { id: 'c', allowed_parent_ids: ['a', 'b'] }
+            ]
+          }, {
+            label: 'an already nested simple nested graph',
+            starting_graph: { parent_ids: { a: [], b: ['a'], c: ['b'] } },
+            expectations: [
+              { id: 'a', allowed_parent_ids: [] },
+              { id: 'b', allowed_parent_ids: ['a'] },
+              { id: 'c', allowed_parent_ids: ['a', 'b'] }
```

Comment:
If I'm reading this correctly, a is already a parent of b and b is already a parent of c.  I was under the impression that existing ancestors are not allowed to be parents.  So is `allowed_parent_ids` identifying the ancestors that already exist for an id in this graph OR the acceptable potential parents?

If the later, I would have expected...
```
{ id: 'a', allowed_parent_ids: [] },       # because b and c are descendants and can't be a's parent
{ id: 'b', allowed_parent_ids: [] },       # because c is a descendant and a is already a parent 
{ id: 'c', allowed_parent_ids: [] }        # because a and b are ancestors
```

I interpret the starting_graph as:  a -> b -> c

I need to understand this to complete the review.

Reply:
I think I see what this is testing.  It is testing whether the indexer rejects a node that cannot be indexed as a parent of the current node.  By this definition, all existing ancestors can and will be in the parent path.  Also, all nodes that are not descendants of the current node can be in the parent path.

@jeremyf Is this interpretation correct?

Reply:
@elrayle Your second comment is correct. But I need to clarify my intentions in the code. The intention is to demonstrate how you can query valid collections that the item can be part of by way of the pathnames. I'll amend the PR to include further clarification.